streamy.py

```diff
@@ -75,3 +56,60 @@ def stream(fp, json_lines=False, **kwds):
             obj = json.loads(json_str, **kwds)
             start_pos += end_pos
             yield obj
+
+
+def _stream_without_seek(fp, kwds):
+    buf = ''
+    # XXX: alternatively could read a limited size buffer
+    for l in fp:
+        buf += l
+        while buf:
+            try:
+                obj = json.loads(buf, **kwds)
+            except ValueError as exc:
+                end_pos = _get_end_pos(exc)
+                if end_pos is None:
+                    if exc.msg.startswith('Expecting value') \
+                       and not buf.strip():
+                        buf = ''
+                        continue
+                    else:
+                        raise
+                obj = json.loads(buf[:exc.pos])
+                buf = buf[exc.pos:]
+            else:
+                buf = ''
+            yield obj
+    if buf:
+        raise ValueError('Extra data!')
+
+
+def stream(fp, json_lines=False, **kwds):
+    """
+    A function generating a stream of valid JSON objects.
+
+    Args:
+        fp: a file stream like you'd get from `open()` or `io.StringIO()`,
+            or a string.
+        json_lines: if true, each line holds at most one JSON expression.
```

Comment:
Yeah, this is a crappy name (which I wrote :-D).  If you have a better name, I'm all ears.